Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/PredictorCorrector.java

```diff
@@ -271,27 +270,43 @@ protected boolean trigger(DiVector candidate, int difference, int baseDimension,
         if (lastAnomalyAttribution == null || difference >= dimensions) {
             return true;
         }
-        checkArgument(lastAnomalyAttribution.getDimensions() == dimensions, " error in DiVectors");
+        double lastAnomalyScore = lastAnomalyDescriptor.getRCFScore();
+        double differentialRemainder = 0;
+        for (int i = dimensions - difference; i < dimensions; i++) {
+            differentialRemainder += Math.abs(candidate.low[i] - ideal.low[i])
+                    + Math.abs(candidate.high[i] - ideal.high[i]);
+        }
+        return (differentialRemainder > DEFAULT_DIFFERENTIAL_FACTOR * lastAnomalyScore)
+                && differentialRemainder * dimensions / difference > workingThreshold;
```

Comment:
Your observation is correct - but difference is bounded by dimensions (the size of the shingle). What the code checks is that the contribution of the new part of the shingle extrapolated to overall shingle.

Reply:
I still don't understand the intuition behind the harder to trigger an anomaly the further out it is from the last anomaly. Will ask you offline.

Reply:
I get it -- your question is "does it make it 'harder' to trigger "? The answer is not really -- it more or less keeps the same threshold but scales up the new component. This could be an issue to dig deeper in the next PR and actually make it 'harder' by raising the trigger. I believe in prior versions the trigger was set higher -- need to have some measurements re how this affects number of anomalies. In the next PR the goal would be to tag/eliminate drifts/runs of anomalies -- so this could come in handy.

Reply:
Got it. I noticed that the differential remainder is being scaled by dimensions / difference. However, I believe this scaling factor may not be correctly representing the "complete" difference if we are only considering a part of the array (difference to dimensions-1).

Should we consider scaling the differential remainder by dimensions / (dimensions - difference) instead? This would ensure that the differential remainder correctly represents the proportion of the total range we're considering when compared to the workingThreshold.